### PR TITLE
feat(jax): build nlist in the SavedModel & fix nopbc for StableHLO and SavedModel

### PR DESCRIPTION
**Pull request by njzjz**
Per our discussion, use TF to build the neighbor list in the SavedModel format.

Also, fix a bug when the number of ghost atoms is zero. The polymorphic_shape needs to be larger than 1, and `nghost == 0` triggered the error. Previously, I also tried `nall` or `nghost - 1` but none of them worked. Finally, I export two different functions... So now four functions are stored in the model: calculate virial or not, x nghost is zero or not. The tests for nopbc are added.

## Release Notes

- **New Features**
	- Enhanced model initialization with additional parameters for improved functionality.
	- Introduced functions for neighbor list management and region transformations in molecular simulations.
	- Added new methods for handling atomic virial calculations in model predictions.
	- New functions for transforming model outputs to accommodate local and ghost atoms.

- **Bug Fixes**
	- Improved error handling in model serialization and evaluation processes.

- **Tests**
	- Added comprehensive unit tests for new functionalities, ensuring consistent behavior across different scenarios, including tests for neighbor list construction and region transformations.

- **Chores**
	- Updated testing workflow for better organization and efficiency.
	- Modified dependency management and linting configurations in `pyproject.toml`.

### CODE REVIEW

File: deepmd/jax/jax2tf/nlist.py
Comment (wanghan-iapcm) [inline deepmd/jax/jax2tf/nlist.py:1]: could you please transfer all the tests for nlist and region to tf?

**Review by wanghan-iapcm — CHANGES_REQUESTED** [review]

**Comment by njzjz** [comment]
The test failed due to out-of-memory. I don't know why JAX consumed so much memory... I add a fixture to call `gc.collect()` for each package.

**Comment by njzjz** [comment]
> The test failed due to out-of-memory. I don't know why JAX consumed so much memory... I add a fixture to call `gc.collect()` for each package.

Oh, it does not work.

**Comment by njzjz** [comment]
I move the JAX test in `test_io.py` into a separated run...

**Comment by njzjz** [comment]
It finally passed after I moved the `tf2jax` directory out of `tests`... I don't know why

**Review by wanghan-iapcm — APPROVED** [review]

**Pull request merged**